src/otx/algo/visual_prompting/openvino_models.py

```diff
@@ -0,0 +1,151 @@
+        self.has_mask_input = np.zeros((1, 1), dtype=np.float32)
+
+    @classmethod
+    def parameters(cls) -> dict[str, Any]:  # noqa: D102
+        parameters = super().parameters()
+        parameters.update({"image_size": NumericalValue(value_type=int, default_value=1024, min=0, max=2048)})
+        parameters.update({"mask_threshold": NumericalValue(value_type=float, default_value=0.0, min=0, max=1)})
+        parameters.update({"embed_dim": NumericalValue(value_type=int, default_value=256, min=0, max=512)})
+        parameters.update({"embedded_processing": BooleanValue(default_value=True)})
+        return parameters
+
+    def _get_outputs(self) -> str:
+        return "upscaled_masks"
+
+    def preprocess(self, inputs: dict[str, Any]) -> list[dict[str, Any]]:
+        """Preprocess prompts."""
+        processed_prompts: list[dict[str, Any]] = []
+        idx: int = 0
+        for prompt_name in ["bboxes", "points"]:
+            prompts = inputs.get(prompt_name, None)
+            if prompts is None:
+                continue
+            for prompt in prompts:
+                label = inputs["labels"][idx]
+                if prompt_name == "bboxes":
+                    point_coords = self.apply_coords(prompt.reshape(-1, 2, 2), inputs["orig_size"])
+                    point_labels = np.array([2, 3], dtype=np.float32).reshape(-1, 2)
+                else:
+                    point_coords = self.apply_coords(prompt.reshape(-1, 1, 2), inputs["orig_size"])
+                    point_labels = np.array([1], dtype=np.float32).reshape(-1, 1)
+
+                processed_prompts.append(
+                    {
+                        "point_coords": point_coords,
+                        "point_labels": point_labels,
+                        "mask_input": self.mask_input,
+                        "has_mask_input": self.has_mask_input,
+                        "orig_size": np.array(inputs["orig_size"], dtype=np.int64).reshape(-1, 2),
+                        "label": label,
+                    },
+                )
+                idx += 1
+        return processed_prompts
+
+    def apply_coords(self, coords: np.ndarray, orig_size: np.ndarray | list[int] | tuple[int, int]) -> np.ndarray:
+        """Process coords according to preprocessed image size using image meta."""
+        old_h, old_w = orig_size
+        new_h, new_w = self._get_preprocess_shape(old_h, old_w, self.image_size)
+        coords = deepcopy(coords).astype(np.float32)
+        coords[..., 0] = coords[..., 0] * (new_w / old_w)
+        coords[..., 1] = coords[..., 1] * (new_h / old_h)
+        return coords
+
+    def _get_preprocess_shape(self, old_h: int, old_w: int, image_size: int) -> tuple[int, int]:
+        """Compute the output size given input size and target image size."""
+        scale = image_size / max(old_h, old_w)
+        new_h, new_w = old_h * scale, old_w * scale
+        new_w = int(new_w + 0.5)
+        new_h = int(new_h + 0.5)
```

Comment:
why is this required?

Reply:
`int` does flooring floating point. If the value is 3.9, we can assume it may be 4 for integer, but it's actually 3.